pyispyb/app/extensions/database/definitions.py

```diff
@@ -52,6 +54,42 @@ def get_options() -> Options:
     return app.db_options
 
 
+def authorize_for_proposal(proposalId: int) -> True:
+    query = db.session.query(models.Proposal).filter(
+        models.Proposal.proposalId == proposalId
+    )
+    query = with_authorization_proposal(query)
+    res = query.count()
+    if res == 0:
+        raise HTTPException(
+            status_code=403, detail="User is not authorized for proposal"
```

Comment:
Here i would return 404 Proposal not found and log that the user was attempting to access a proposal for which they didn't have access